token_server/token_server.js

```diff
@@ -0,0 +1,88 @@
+import * as GoogleAuth from 'google-auth-library';
+import {createServer} from 'http';
+// TODO(janakr): this is a pretty random package. Maybe find a more popular one.
+import parseBody from 'urlencoded-body-parser';
+import {CLIENT_ID, getMillisecondsToDateString} from '../docs/common_auth_utils.js';
+
+import {generateEarthEngineToken} from './ee_token_creator.js';
+
+const RESPONSE_HEADERS = {
+  'Content-type': 'text/plain',
+  'Vary': 'Origin',
+};
+
+const ONE_MINUTE_IN_MILLISECONDS = 60 * 1000;
+
+// Regenerate token every 40 minutes, hopefully will be ready whenever a user
```

Comment:
Because 40 minutes < 1 hour - 5 minutes, where 1 hour is token validity and 5 minutes is minimum validity user code will accept. Basically, we want a valid token to be ready whenever the user asks for one.

There was an issue with this code, though: during token regeneration, we weren't keeping around the still-valid current token. Fixed that, and got rid of the paranoid check, since it made things more complicated.